android/src/main/java/com/amplitude/android/Amplitude.kt

```diff
@@ -53,12 +54,28 @@ open class Amplitude(
             lastEventTime = storage.read(Storage.Constants.LAST_EVENT_TIME) ?. let {
                 it.toLong()
             } ?: -1
-            add(AndroidContextPlugin())
+            androidContextPlugin = AndroidContextPlugin()
+            add(androidContextPlugin)
             add(AndroidLifecyclePlugin())
         }
         add(AmplitudeDestination())
     }
 
+    /**
+     * Reset identity:
+     *  - reset userId to "null"
+     *  - reset deviceId via AndroidContextPlugin
+     * @return the Amplitude instance
+     */
+    override fun reset(): Amplitude {
+        this.setUserId(null)
+        amplitudeScope.launch(amplitudeDispatcher) {
+            idContainer.identityManager.editIdentity().setDeviceId(null).commit()
+            androidContextPlugin.initializeDeviceId(configuration as Configuration)
```

Comment:
A good thought for if we still respect the settings. I wonder if in this case we can just call the uuid method and skip the logic to check adid or appset id

Reply:
So we just set the deviceId to uuid without respecting the settings? I am ok with both, but not sure whether this will introduce confusion to users.

Reply:
yeah, I guess that could work. We can also keep the current way and see how it goes